libsolidity/analysis/GlobalContext.cpp

```diff
@@ -35,10 +35,43 @@ namespace dev
 namespace solidity
 {
 
+int magicVariableToID(std::string const& _name)
```

Comment:
After having looked through source finally get the reason why these are negative: to easily differentiate between IDs of elements from a user source. Should document this.